include/RED4ext/Containers/DynArray.hpp

```diff
@@ -0,0 +1,584 @@
+    {
+        return ConstIterator(m_entries + m_size);
+    }
+
+    [[nodiscard]] ReverseIterator RBegin() noexcept
+    {
+        return ReverseIterator(Begin());
+    }
+
+    [[nodiscard]] ConstReverseIterator RBegin() const noexcept
+    {
+        return ConstReverseIterator(Begin());
+    }
+
+    [[nodiscard]] ReverseIterator REnd() noexcept
+    {
+        return ReverseIterator(End());
+    }
+
+    [[nodiscard]] ConstReverseIterator REnd() const noexcept
+    {
+        return ConstReverseIterator(End());
+    }
+
+    [[nodiscard]] bool Empty() const noexcept
+    {
+        return m_size == 0;
+    }
+
+    [[nodiscard]] Pointer Data() noexcept
+    {
+        return m_entries;
+    }
+
+    [[nodiscard]] ConstPointer Data() const noexcept
+    {
+        return m_entries;
+    }
+
+    [[nodiscard]] SizeType Capacity() const noexcept
+    {
+        return m_capacity;
+    }
+
+    [[nodiscard]] constexpr SizeType MaxSize() const noexcept
+    {
+#undef max // avoid conflict with minmax
+#undef min
+        static constexpr auto maxSize =
+            std::min(std::numeric_limits<SizeType>::max(),
+                     static_cast<SizeType>(std::numeric_limits<uint32_t>::max() / sizeof(ValueType)));
+        return maxSize;
+    }
+
+    [[nodiscard]] SizeType Size() const noexcept
+    {
+        return m_size;
+    }
+
+#pragma region STL
+    [[nodiscard]] Iterator begin() noexcept
+    {
+        return Begin();
+    }
+
+    [[nodiscard]] ConstIterator begin() const noexcept
+    {
+        return Begin();
+    }
+
+    [[nodiscard]] Iterator end() noexcept
+    {
+        return End();
+    }
+
+    [[nodiscard]] ConstIterator end() const noexcept
+    {
+        return End();
+    }
+#pragma endregion
+private:
+    T* m_entries;        // 00
+    uint32_t m_capacity; // 08
+    uint32_t m_size;     // 0C
+
+    [[nodiscard]] bool Includes(ConstIterator aPos) const noexcept
+    {
+        return Begin() <= aPos && aPos <= End();
+    }
+
+    [[nodiscard]] bool Includes(ConstIterator aFirst, ConstIterator aLast) const noexcept
+    {
+        return Begin() <= (std::min)(aFirst, aLast) && (std::max)(aLast, aFirst) <= End();
+    }
+
+    template<class InputIt>
+    requires std::same_as<InputIt, Iterator> || std::same_as<InputIt, ConstIterator>
+    void ShiftEntries(InputIt aSrc, InputIt aDst, SizeType aCount) noexcept
+    {
+        if (aCount == 0 || aSrc == aDst)
+            return;
+
+        if (aSrc >= aDst)
+            std::move(aSrc, aSrc + aCount, aDst);
+        else
+            std::move_backward(aSrc, aSrc + aCount, aDst + aCount);
+    }
+
+    static void MoveEntries(Pointer aDstBuffer, Pointer aSrcBuffer, uint32_t aSrcSizeInBytes, DynArray* aSrcArray)
+    {
+        if (aSrcSizeInBytes == 0)
+            return;
+
+        std::move(aSrcBuffer, aSrcBuffer + aSrcArray->Size(), aDstBuffer);
+        std::destroy(aSrcBuffer, aSrcBuffer + aSrcArray->Size());
```

Comment:
It can be `aSrcArray->m_size`.

Reply:
I would say  `Size` should be called here and the class' member fields be only used in their respective methods.